src/screens/lesson/Lesson.js

```diff
@@ -0,0 +1,340 @@
+import React, { useState, useEffect } from "react";
+import { View, Text, TouchableOpacity, Image, ScrollView, Modal } from "react-native";
+import styles from "./styles";
+import HoverableView from "../../compoments/HoverableView";
+import { useNavigate } from "react-router-dom";
+import { AiFillHome, AiOutlineRead, AiOutlineVideoCamera, AiOutlineCheck } from "react-icons/ai";
+import { supabase } from "../../supabaseClient";
+import * as _ from 'lodash';
+
+export default function Lesson(props) {
+  const [subjectItems, setSubjectItems] = useState([]);
+  const [lessonItems, setLessonItems] = useState([]);
+  const [modalVisible, setModalVisible] = useState(false);
+  const [lessonId, setLessonId] = useState(0);
+  const [lessonName, setLessonName] = useState(null);
+
+  let navigate = useNavigate();
+  const goToCourseScreen = () => {
+    navigate("/");
+  };
+
+  const onShowContent = (urlContent) => {
+    window.open(`${urlContent}`, '_blank');
+  };
+
+  const getSubjectItems = async () => {
+    const { data, error } = await supabase
+      .from('subjects')
+      .select('*, lessons!inner(*)')
+
+    if(error) {
+      throw new Error(error.message)
+    }
+    if(!data) {
+      throw new Error("Subjects not found")
+    }
+    setSubjectItems(data);
+  };
+  
+  useEffect(() => {
+    getSubjectItems();
+  });
+
+  const getLessonItems = async () => {
+    const { data, error } = await supabase
+      .from('lessons')
+      .select('*, courses!inner(*)')
+      .filter('courses.id', 'in', `(${props.courseId})`)
```

Comment:
Can we use equal operator over here? FYI equal will be faster than in operator

Reply:
Edited
https://github.com/cam-drei/kids-study-front-end/pull/1/commits/f215e9ab6cdb1b900dac8db67587e9ba0b03a7ef